cppcache/include/geode/CacheableBuiltins.hpp

```diff
@@ -158,20 +158,17 @@ class APACHE_GEODE_EXPORT CacheableContainerType
 #pragma warning(disable : 4231)
 #endif
 
-#define _GEODE_CACHEABLE_KEY_TYPE_DEF_(p, k)                      \
-  extern const char tName_##k[];                                  \
+#define _GEODE_CACHEABLE_KEY_TYPE_DEF_(p, k)                \
+  extern const char tName_##k[];                            \
   template class CacheableKeyType<p, DSCode::k, tName_##k>; \
   typedef CacheableKeyType<p, DSCode::k, tName_##k> _##k;
 
 // use a class instead of typedef for bug #283
 #define _GEODE_CACHEABLE_KEY_TYPE_(p, k)                                \
   class APACHE_GEODE_EXPORT k : public _##k {                           \
-   protected:                                                           \
+   public:                                                              \
     inline k() : _##k() {}                                              \
```

Comment:
Unless the default constructor was used by one of the factory methods then it was likely protected only to "delete" it before `delete` was available. Consider using `delete` if it is not used.

This applies to all other default constructors in this change.